src/sagas/projects.js

```diff
@@ -47,6 +50,17 @@ export function* changeCurrentProject() {
   yield call(saveCurrentProject, state);
 }
 
+export function* rehydrateProject({payload: {rehydratedProject}}) {
+  const state = yield select();
+  const allProjectKeys = getProjectKeys(state);
+  if (allProjectKeys.includes(rehydratedProject.projectKey)) {
+    yield put(__changeCurrentProject(rehydratedProject.projectKey));
+  } else {
+    yield put(projectLoaded(rehydratedProject));
+    yield put(__changeCurrentProject(rehydratedProject.projectKey));
```

Comment:
@ajgreenb nope, what I’m proposing is a bit different / simpler I think:

The `rehydrateProject()` saga dispatches `PROJECT_RESTORED_FROM_LAST_SESSION` by `put`ting `projectRestoredFromLastSession(rehydratedProject)`. Regardless of login state, the reducers and sagas handle this action appropriately, to wit:

* The `currentProject` reducer updates the `currentProject.projectKey` to be that of the rehydrated project
* The `projects` reducer adds the project to the collection of projects
* The `errors` saga initiates validation

Effectively the system would react to `PROJECT_RESTORED_FROM_LAST_SESSION` in the same way it reacts to `SNAPSHOT_IMPORTED`. However, there’s nothing fundamental that _requires_ that the system react in the same way to those two actions, so it’s best to keep them as separate actions to allow more differences to emerge in the future.